File: TASC-Intern-Batch9/pharmacy/src/main/java/com/project/pharmacy/service/ProfileService.java
package com.project.pharmacy.service;

import com.project.pharmacy.dto.request.ProfileRequest;
import com.project.pharmacy.dto.response.ApiResponse;
import com.project.pharmacy.dto.response.ProfileResponse;

import java.util.List;

public interface ProfileService {
    ApiResponse<List<ProfileResponse>> getUserProfiles();
    ApiResponse<ProfileResponse> createProfile(ProfileRequest request);
    ApiResponse<ProfileResponse> updateProfile(Long id, ProfileRequest request);
    ApiResponse<Void> deleteProfile(Long id);
    ApiResponse<ProfileResponse> getProfileById(Long id);
}

File: TASC-Intern-Batch9/fe/fe-client/src/api/profileService.ts
import axiosClient from './axiosClient';
import type { ApiResponse, Profile } from '../types';
import type { ProfileRequest } from '../types/profile.types';

const profileService = {
  // Lấy danh sách địa chỉ
  getUserProfiles: () => {
    return axiosClient.get<ApiResponse<Profile[]>>('/profiles');
  },

  // Lấy chi tiết 1 địa chỉ
  getProfileById: (id: number) => {
    return axiosClient.get<ApiResponse<Profile>>(`/profiles/${id}`);
  },

  // Tạo mới
  createProfile: (data: ProfileRequest) => {
    return axiosClient.post<ApiResponse<Profile>>('/profiles', data);
  },

  // Cập nhật
  updateProfile: (id: number, data: ProfileRequest) => {
    return axiosClient.put<ApiResponse<Profile>>(`/profiles/${id}`, data);
  },

  // Xóa
  deleteProfile: (id: number) => {
    return axiosClient.delete<ApiResponse<void>>(`/profiles/${id}`);
  }
};

export default profileService;